models/catalog.cattle.io.clusterrepo.js

```diff
@@ -29,6 +29,21 @@ export default class ClusterRepo extends SteveModel {
     return out;
   }
 
+  get customValidationRules() {
+    const out = [];
+
+    if (this.isGit) {
```

Comment:
`this.isGit` is only true if `value.spec.gitRepo` is non-empty, so we're still able to make repos with neither a git repo url nor git branch specified despite both being marked as required. I think we need a bit fancier logic around disabling the create button on this page, where either `spec.url` or `spec.gitRepo` and `spec.gitBranch` must be defined.

Reply:
Agreed 100% although the backend would kick back an error if you sent it a payload with neither url or gitRepo defined. You can also define both in the current UI without being able to see both fields at the same time to know what the problem is. It's probably outside of the scope of this ticket to completely rewrite and redesign the form though.

Pulling in @lvuch to define what we'd expect from the UX in this particular case.